src/Components/CreateImageWizardV2/ImageWizardContext.tsx

```diff
@@ -0,0 +1,208 @@
+    filterEnvironment(
+      {
+        aws: { selected: false, authorized: false },
+        azure: { selected: false, authorized: false },
+        gcp: { selected: false, authorized: false },
+        oci: { selected: false, authorized: false },
+        'vsphere-ova': { selected: false, authorized: false },
+        vsphere: { selected: false, authorized: false },
+        'guest-image': { selected: false, authorized: false },
+        'image-installer': { selected: false, authorized: false },
+        wsl: { selected: false, authorized: false },
+      },
+      allowedTargets
+    )
+  );
+  // Update of the environment when the architecture and release are changed.
+  // This pattern prevents the usage of a useEffect See https://react.dev/learn/you-might-not-need-an-effect#adjusting-some-state-when-a-prop-changes
+  const [prevAllowedTargets, setPrevAllowedTargets] = useState(allowedTargets);
+  if (!isIdenticalToPrev(prevAllowedTargets, allowedTargets)) {
+    setPrevAllowedTargets(allowedTargets);
+    setEnvironment(filterEnvironment(environment, allowedTargets));
+  }
+
+  //
+  // Target environment states
+  //
+  // AWS
+  const [awsManual, setAwsManual] = useState(false);
+  const [awsSource, setAwsSource] = useState<[number, string]>([0, '']);
+  const { accountId: awsID } = useGetAccountData(awsSource[0], 'aws');
+  const [awsAccountId, setAwsAccountId] = useState(awsID);
+  const [prevAwsID, setPrevAwsID] = useState(awsID);
+  if (awsID !== prevAwsID) {
+    setPrevAwsID(awsID);
+    setAwsAccountId(awsID);
+  }
+  // GCP
+  const [shareGoogleAccount, setShareGoogleAccount] = useState(true);
+  const [gcpAccountType, setGcpAccountType] =
+    useState<GCPAccountTypes>('googleAccount');
+  const [gcpAccountEmail, setGcpAccountEmail] = useState('');
+  const [gcpDomain, setGcpDomain] = useState('');
+  // Azure
+  const [azureManual, setAzureManual] = useState(false);
+  const [azureSource, setAzureSource] = useState<[number, string]>([0, '']);
+  const { tenantId, subscriptionId } = useGetAccountData(
+    azureSource[0],
+    'azure'
+  );
+  const [azureTenantId, setAzureTenantId] = useState(tenantId);
+  const [azureSubId, setAzureSubId] = useState(subscriptionId);
+  const [azureResourceGroup, setAzureResourceGroup] = useState('');
+  const [prevTenantId, setPrevTenantId] = useState(tenantId);
+  const [prevSubscriptionId, setPrevSubscriptionId] = useState(subscriptionId);
+  if (prevTenantId !== tenantId || prevSubscriptionId !== subscriptionId) {
+    setPrevTenantId(tenantId);
+    setPrevSubscriptionId(subscriptionId);
+    setAzureTenantId(tenantId);
+    setAzureSubId(subscriptionId);
+    // when the tenant id or the subscription_id is changing, reset the resource
+    // group as the user will need to update it from a new list.
+    setAzureResourceGroup('');
+  }
+  const context: ImageWizardContextPropType = {
+    releaseState: [release, setRelease],
```

Comment:
I'm not quite sold on this. Using an array to pass the state value and setters around seems like an anti-pattern to me. When people read the code who aren't familiar with this, it will be unclear what we are doing when accessing releaseState[0] and releaseState[1]. It also means we aren't taking advantage of the type safety Typescript is giving us now. Would the linter pick up on someone trying to access releaseState[2]? Also, I don't think that the Typescript language server is going to be able to make meaningful inferences/suggestions. 

I'll try to look into some alternate means of handling this.

Reply:
It's not an array (well it looks like it lol) but a tupple for typescript. It considers it an array that has two members an not more with a specific order in which the values must be passed in. But yeah, if you want to try something else that's fine too.

Reply:
Ah sorry I wasn't on the right one. Forget my comment.